Assets/Scripts/Gameplay/GameplayObjects/Character/ServerCharacter.cs

```diff
@@ -1,30 +1,110 @@
+using System;
 using System.Collections;
-using Unity.Multiplayer.Samples.BossRoom.Actions;
+using Unity.BossRoom.Gameplay.Actions;
+using Unity.BossRoom.Gameplay.Configuration;
+using Unity.BossRoom.Gameplay.GameplayObjects.Character.AI;
+using Unity.Multiplayer.Samples.BossRoom;
 using Unity.Netcode;
 using UnityEngine;
+using Action = Unity.BossRoom.Gameplay.Actions.Action;
 
-namespace Unity.Multiplayer.Samples.BossRoom.Server
+namespace Unity.BossRoom.Gameplay.GameplayObjects.Character
 {
-    public class ServerCharacter : NetworkBehaviour
+    /// <summary>
+    /// Contains all NetworkVariables, RPCs and server-side logic of a character.
+    /// This class was separated in two to keep client and server context self contained. This way you don't have to continuously ask yourself if code is running client or server side.
+    /// </summary>
+    [RequireComponent(typeof(NetworkHealthState),
+        typeof(NetworkLifeState),
+        typeof(NetworkAvatarGuidState))]
+    public class ServerCharacter : NetworkBehaviour, ITargetable
     {
         [SerializeField]
-        NetworkCharacterState m_NetworkCharacterState;
+        ClientCharacter m_ClientVisualization;
 
-        public NetworkCharacterState NetState => m_NetworkCharacterState;
+        public ClientCharacter ClientVisualization => m_ClientVisualization;
```

Comment:
Should this still be "ClientVisualization"?

Reply:
ClientCharacter it is then